### PR TITLE
Migrate reactions when migrating repository from github

### PR DESCRIPTION
**Pull request by lunny**
- [x] Template change, don't think migrator is the reaction author.
- [x] Unit tests
- [x] Update original author id when user connected external user account.

An issue migrated from https://github.com/go-gitea/test_repo/issues/2 as below:

图片

### CODE REVIEW

**Comment by lunny** [comment]
blocked by #9693

File: models/issue_reaction.go [hunk 1]
```diff
@@ -233,7 +235,7 @@ func (list ReactionList) HasUser(userID int64) bool {
 		return false
 	}
 	for _, reaction := range list {
-		if reaction.UserID == userID {
+		if reaction.OriginalAuthor == "" && reaction.UserID == userID {
```
[the remark this answers is not in the document]
Comment (lunny) [reply models/issue_reaction.go:241]: When a reaction's user is an external one, `OriginalAuthor` will not be empty and `UserID` is only the migrator user id.

[the remark this answers is not in the document]
Comment (guillep2k) [reply models/issue_reaction.go:241]: Right, sorry.

**Review by guillep2k — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Comment by sapk** [comment]
Oh sorry @6543, I didn't see your comment.

**Comment by sapk** [comment]
Can you confirm that it was fixed by https://github.com/go-gitea/gitea/pull/9599/commits/6ac563a8597b03d179382dfca9a21358910279dc ?

**Pull request merged**